File: client_packages/gamemode/fly.js
"use strict";
var rotation = new mp.Vector3(0, 0, 0);
var isFly = false;
var player = mp.players.local;
var rotation = new mp.Vector3(0, 0, 0);
var speed = 0.5;
var keymap = {
    w: 87,
    s: 83,
    a: 65,
    d: 68,
    q: 81,
    e: 69,
    lshift: 16,
    lctrl: 17
};
mp.events.add('toggle_fly', () => {
    isFly = !isFly;
    player.setCollision(!isFly, !isFly);
    player.freezePosition(isFly);
    mp.gui.chat.push(`fly ${isFly}`);
});
mp.events.add('render', () => {
    if (isFly) {
        if (mp.keys.isDown(keymap.lshift)) {
            speed = 2;
        }
        else if (mp.keys.isDown(keymap.lctrl)) {
            speed = 0.2;
        }
        else
            speed = 0.5;
        if (mp.keys.isDown(keymap.w)) {
            player.setCoordsNoOffset(player.position.x, player.position.y + speed, player.position.z, true, true, true);
        }
        if (mp.keys.isDown(keymap.s)) {
            player.setCoordsNoOffset(player.position.x, player.position.y - speed, player.position.z, true, true, true);
        }
        if (mp.keys.isDown(keymap.a)) {
            player.setCoordsNoOffset(player.position.x + speed, player.position.y, player.position.z, true, true, true);
        }
        if (mp.keys.isDown(keymap.d)) {
            player.setCoordsNoOffset(player.position.x - speed, player.position.y, player.position.z, true, true, true);
        }
        if (mp.keys.isDown(keymap.q)) {
            player.setCoordsNoOffset(player.position.x, player.position.y, player.position.z - speed, true, true, true);
        }
        if (mp.keys.isDown(keymap.e)) {
            player.setCoordsNoOffset(player.position.x, player.position.y, player.position.z + speed, true, true, true);
        }
    }
});
mp.events.add('render', () => {
    mp.game.graphics.drawText(`X: ${player.position.x}`, [0.5, 0.005], {
        font: 7,
        color: [255, 255, 255, 185],
        scale: [0.5, 0.5],
        outline: true
    });
    mp.game.graphics.drawText(`Y: ${player.position.y}`, [0.5, 0.05
    ], {
        font: 7,
        color: [255, 255, 255, 185],
        scale: [0.5, 0.5],
        outline: true
    });
    mp.game.graphics.drawText(`Z: ${player.position.z}`, [0.5, 0.1], {
        font: 7,
        color: [255, 255, 255, 185],
        scale: [0.5, 0.5],
        outline: true
    });
    mp.game.graphics.drawText(`RotX: ${player.getRotation(0).x}`, [0.25, 0.1], {
        font: 7,
        color: [255, 255, 255, 185],
        scale: [0.5, 0.5],
        outline: true
    });
});

File: client_packages/cs_packages/uitl/Fly.cs
using System.Collections.Generic;
using RAGE;
using RAGE.Elements;
using RAGE.Ui;

namespace ClientSide.uitl
{
    public class Fly : Events.Script
    {
        private bool isFly = false;
        private double speed = 0.5;
        private Player _player = RAGE.Elements.Player.LocalPlayer;
        
        public Fly()
        {
            Events.Tick += render;
            RAGE.Events.Add("toggle_fly", args =>
            {
                isFly = !isFly;
                _player.FreezePosition(isFly);
            });
        }
        
        private void render(List<Events.TickNametagData> TickNametags)
        {
            if (isFly)
            {
                if (RAGE.Input.IsDown(VirtualKeys.LeftShift))
                {
                    speed = 2;
                } 
                else if (RAGE.Input.IsDown(VirtualKeys.LeftControl))
                {
                    speed = 0.2;
                }
                else speed = 0.5;

                if (RAGE.Input.IsDown(VirtualKeys.W))
                {
                    _player.SetCoordsNoOffset(_player.Position.X, _player.Position.Y + (float) speed, _player.Position.Z, true,true,true);
                }
                if (RAGE.Input.IsDown(VirtualKeys.S))
                {
                    _player.SetCoordsNoOffset(_player.Position.X, _player.Position.Y - (float) speed, _player.Position.Z, true,true,true);
                }
                if (RAGE.Input.IsDown(VirtualKeys.A))
                {
                    _player.SetCoordsNoOffset(_player.Position.X - (float) speed, _player.Position.Y, _player.Position.Z, true,true,true);
                }
                if (RAGE.Input.IsDown(VirtualKeys.D))
                {
                    _player.SetCoordsNoOffset(_player.Position.X + (float) speed, _player.Position.Y, _player.Position.Z, true,true,true);
                }
                if (RAGE.Input.IsDown(VirtualKeys.E))
                {
                    _player.SetCoordsNoOffset(_player.Position.X, _player.Position.Y, _player.Position.Z + (float) speed, true,true,true);
                }
                if (RAGE.Input.IsDown(VirtualKeys.Q))
                {
                    _player.SetCoordsNoOffset(_player.Position.X, _player.Position.Y, _player.Position.Z - (float) speed, true,true,true);
                }
            }    
        }
    }
}